cdap-ui/app/features/hydratorplusplus/services/non-store-pipeline-error-factory.js

```diff
@@ -126,7 +126,7 @@ let isNodeNameUnique = (myHelpers, nodeName, nodes, cb) => {
 
 let hasValidName = (name, cb) => {
   let error;
-  let pattern = /^[\w]+$/;
+  let pattern = /^[\w-]+$/;
```

Comment:
Hmm, nevermind.
http://www.w3schools.com/jsref/jsref_regexp_wordchar.asp
"A word character is a character from a-z, A-Z, 0-9, including the _ (underscore) character."